.github/workflows/ci.yml

```diff
@@ -0,0 +1,68 @@
+name: ci
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.7", "3.8", "3.9", "3.10"]
+        poetry-version: [1.1.0]
+        os: [ubuntu-18.04, macos-latest, windows-latest]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install poetry
+        run: pipx install poetry
+      - uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+          cache: "poetry"
+      - name: Install updatebot dependencies
+        run: poetry install
+  lint:
+    needs: build
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.7"]
+        poetry-version: [1.1.0]
+        os: [ubuntu-18.04]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install poetry
+        run: pipx install poetry
```

Comment:
Do we still need to checkout and install pip and setup python here (and below) if it's in the build task we depend on?

Reply:
The caching from the build step is for dependencies only, sadly, so we do still need to do that. each of the jobs runs in its own container using either github's `ubuntu-latest` or `windows-latest` image.